### ISSUE TITLE
Should catch blocks be nested inside the  try block

### ISSUE THREAD

**Issue by KarlSchimpf** [WebAssembly/exception-handling#5]
The current proposal puts the catch blocks inside the try block. This makes it clear what the enclosing try  block is. However, it makes the description of throws within catch blocks more complex.

Would a better approach be to have the catch blocks immediately follow the try block?

**Comment by rossberg-chromium** [comment]
I would assume that `catch` isn't a separate block instruction at all, but part of the `try`, in the same way `else` is part of `if`. Roughly, the syntax would be:
```
try <blocktype> <instr>* (catch <idx> <instr>*)* (default <instr>*) end
```

**Comment by KarlSchimpf** [comment]
I agree with this suggestion. It is cleaner. The only (possible) difference is using "else" instead of "default". I will work on a PR to change this.
